src/__global.zig

```diff
@@ -59,16 +59,61 @@ else if (Environment.isAarch64)
 else
     "unknown";
 
-pub inline fn getStartTime() i128 {
-    if (Environment.isTest) return 0;
-    return @import("root").bun.start_time;
-}
-
-pub fn setThreadName(name: StringTypes.stringZ) void {
-    if (Environment.isLinux) {
-        _ = std.os.prctl(.SET_NAME, .{@intFromPtr(name.ptr)}) catch 0;
-    } else if (Environment.isMac) {
-        _ = std.c.pthread_setname_np(name);
+/// Like std.Thread.setName but auto-truncate, and only works on the current thread.
+/// The u8 -> u16 conversion is ascii-only
+pub fn setThreadName(name: [:0]const u8) void {
+    var truncated_name = switch (bun.OSPathSlice) {
+        [:0]const u8 => if (name.len > std.Thread.max_name_len) brk: {
+            var truncated_name: [std.Thread.max_name_len + 1]u8 = undefined;
```

Comment:
this buffer is stack allocated in this block. is it safe to use it later in this function?